docs/node/web3.md

```diff
@@ -0,0 +1,201 @@
+Install it by following the instructions specific to your operating system. You
+will need to create a separate database for each ParaTime. We also suggest that
+you create and use a different username for connecting to the database for each
+ParaTime.
+
+### Oasis ParaTime Client Node
+
+The Web3 gateway requires a locally deployed ParaTime-enabled Oasis Node.
+First, follow the
+[Oasis ParaTime Client Node](run-your-node/paratime-client-node.mdx)
+guide on how to configure the Oasis client node with one or more ParaTimes.
+Make sure you check the network parameters page ([Mainnet], [Testnet]) and use
+the exact combination of the Oasis node/ParaTime versions!
+
+Next, we suggest that you slightly modify the ParaTime section of your Oasis
+node configuration. By default, the Emerald and Sapphire ParaTimes prevent
+you from executing *expensive* transactions such as simulating EVM calls
+without paying for gas. If your Oasis node instance will only be used by the
+Web3 gateway(s) and otherwise not exposed to the internet, we suggest that you
+enable expensive transactions by adding the following to your Oasis node
+config:
+
+```yaml title='config.yml'
+# ... sections not relevant are omitted ...
+runtime:
+  mode: client
+  paths:
+    - <emerald_bundle_path>
+    - <sapphire_bundle_path>
+
+  config:
+    # Use the following IDs:
+    # 000000000000000000000000000000000000000000000000e2eaa99fc008f87f # Emerald on Mainnet
+    # 00000000000000000000000000000000000000000000000072c8215e60d5bca7 # Emerald on Testnet
+    # 000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c # Sapphire on Testnet
+    "<emerald_paratime_id>":
+      estimate_gas_by_simulating_contracts: true
+      allowed_queries:
+        - all_expensive: true
+    "<sapphire_paratime_id>":
+      estimate_gas_by_simulating_contracts: true
+      allowed_queries:
+        - all_expensive: true
+```
+
+## Download the Web3 Gateway
+
+Check the required version of the Web3 gateway based on which network you will
+be using it with ([Mainnet], [Testnet]). Next, download the Oasis-provided
+binaries from the [official github repository][github-releases].
+
+Alternatively, you can download the source release and compile it yourself.
+Consult the [README] file for more information.
+
+## Running the Gateway on Testnet/Mainnet
+
+Copy the content below to the config file of your gateway and adapt it
+accordingly to fit your needs:
+
+```yaml title='gateway.yml'
+# Uncomment the runtime_id below.
+#runtime_id: "000000000000000000000000000000000000000000000000e2eaa99fc008f87f" # Emerald on Mainnet
+#runtime_id: "00000000000000000000000000000000000000000000000072c8215e60d5bca7" # Emerald on Testnet
+#runtime_id: "000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c" # Sapphire on Testnet
+# Path to your internal.sock file in the root Oasis node datadir.
+node_address: "unix:<path-to-oasis-node-unix-socket>"
```

Comment:
This requires running services in Docker containers or under the same Linux user (see https://github.com/oasisprotocol/oasis-core/issues/4850). There is also a debug parameter that can set up the Unix socket to be outside of the root Oasis node datadir `debug.grpc.internal.socket_path`.